RELEASE_PROCEDURE.md

```diff
@@ -55,4 +55,6 @@ DLA-Future follows [Semantic Versioning](https://semver.org).
 
 1. Delete your `GITHUB_TOKEN` if created only for the release.
 
+1. Create a new milestone named `vX.Y.Z` for the next major or minor release.
+
```

Comment:
Do we want to be more explicit here and mention something about planning what to include in the next release as well? I don't think it necessarily fits, but I don't mind adding it either.